### PR TITLE
Issue 47 preact testing

### PR DESCRIPTION
**Pull request by desdemonhu**
## Issue Description: 
Setup preact testing

### Summary of solution:
1. Added modules mocha, chai, preact-jsx-chai, and ignore-styles
1.a. Set up client package.json file to run npm test config and added needed babel plugin
2. Set up components.test.js as a test for Header Component as a template for other tests
3. Added test-client script to server package.json file and set so that it runs concurrently with server test
Notes: I got an error trying to run the server test, but the client test passed as expected

### Can this issue be closed? 
I think so

### Should any new issues be added as a result of this solution? 
Adding actual tests

### Have you named your branch in a descriptive way? Remember to name your branch in a unique and descriptive manner in order to properly reflect the issue or feature.
^__^d

### CODE REVIEW

**Comment by motosharpley** [comment]
@CodeDraken can you take a look and see if you can spot why she got an error on the server test?
thanks!

**Comment by CodeDraken** [comment]
Yeah

@desdemonhu can you copy/paste any errors you got from running the server tests?

**Comment by desdemonhu** [comment]
I started mongod, but it's possible I did something wrong - I just assumed I had set something wrong in my dev environment. Here is my console log when I run the tests:

`> googlemaps-offline-navigator@0.0.1 test C:\Users\Raegan\repos\googleMaps-offline-navigator
> concurrently "npm run test-server" "npm run test-client"

[0]
[0] > googlemaps-offline-navigator@0.0.1 test-server C:\Users\Raegan\repos\googleMaps-offline-navigator
[0] > NODE_ENV=test mocha tests -r chai/register-expect
[0]
[1]
[1] > googlemaps-offline-navigator@0.0.1 test-client C:\Users\Raegan\repos\googleMaps-offline-navigator
[1] > npm test --prefix client
[1]
[0] 'NODE_ENV' is not recognized as an internal or external command,
[0] operable program or batch file.
[0] npm ERR! code ELIFECYCLE
npm ERR! errno 1
[0] npm ERR! googlemaps-offline-navigator@0.0.1 test-server: `NODE_ENV=test mocha tests -r chai/register-expect`
npm ERR! Exit status 1
[0] npm ERR!
npm ERR! Failed at the googlemaps-offline-navigator@0.0.1 test-server script.
[0] npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
[0]
[0] npm ERR! A complete log of this run can be found in:
[0] npm ERR!     C:\Users\Raegan\AppData\Roaming\npm-cache\_logs\2018-01-24T05_36_56_198Z-debug.log
[0] npm run test-server exited with code 1
[1]
[1] > client@0.0.0 test C:\Users\Raegan\repos\googleMaps-offline-navigator\client
[1] > mocha ./tests --require babel-polyfill --require ignore-styles --compilers js:babel-core/register --reporter spec || exit 0
[1]
[1]
[1]
[1] (node:5932) DeprecationWarning: "--compilers" will be removed in a future version of Mocha; see https://git.io/vdcSr for more info
[1]   Header
[1]     √ passes
[1]
[1]
[1]   1 passing (17ms)
[1]
[1] npm run test-client exited with code 0
npm ERR! Test failed.  See above for more details.
`

**Comment by motosharpley** [comment]
did you run npm install in both client and root since pulling latest version?

**Comment by CodeDraken** [comment]
I think the issue is the `NODE_ENV=test` part. I noticed she is on Windows and I didn't think about that... :thinking: Setting Node variables works differently on each platform. cross-env might be the solution.

https://www.npmjs.com/package/cross-env

**Comment by motosharpley** [comment]
Ya I am on windows too, that would explain why it works when I run the command directly. I will try making that change and see what happens.

File: package.json [hunk 1]
```diff
@@ -8,8 +8,10 @@
     "dev": "concurrently \"npm run server\" \"npm run client\"",
     "client": "npm run start --prefix client",
     "server": "nodemon server.js",
-    "test": "NODE_ENV=test mocha tests -r chai/register-expect",
-    "test-watch": "nodemon --exec 'npm test'"
+    "test": "concurrently \"npm run test-server\" \"npm run test-client\"",
+    "test-server": "NODE_ENV=test mocha tests -r chai/register-expect",
```
Comment (motosharpley) [inline package.json]: install cross-env ...
npm install --save-dev cross-env
change to "test-server": "cross- env NODE_ENV=test mocha tests -r chai/register-expect"

**Comment by desdemonhu** [comment]
Perfect, everything runs for me now, and I pushed the updates to my branch

**Pull request merged**